src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -246,14 +232,22 @@ impl KvmVcpu {
             .map_err(KvmVcpuConfigureError::NormalizeCpuidError)?;
 
         // Set CPUID.
-        // let kvm_cpuid = kvm_bindings::CpuId::from(joined_cpuid);
         let kvm_cpuid = kvm_bindings::CpuId::from(cpuid);
 
         // Set CPUID in the KVM
         self.fd
             .set_cpuid2(&kvm_cpuid)
             .map_err(KvmVcpuConfigureError::SetCpuid)?;
 
+        // Clone MSR entries that are modified by CPU template from `VcpuConfig`.
+        let mut msrs = vcpu_config.cpu_config.msrs.clone();
+        self.msrs_to_save.extend(msrs.keys());
+
+        // Apply MSR modification to comply the linux boot protocol.
```

Comment:
`refactor(vmm): move MSR normalization for boot`
nit: we may need to update the commit message too to avoid `MSR normalization` term.

Reply:
Good catch! thanks.